aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -77,6 +191,23 @@ def data_as_json(self) -> dict:
             self._json_data = self._json_deserializer(self.data_as_text)
         return self._json_data
 
+    def create_firehose_response_record(
+        self,
+        result: Literal["Ok", "Dropped", "ProcessingFailed"],
+        data: Optional[str] = None,
+    ) -> KinesisFirehoseResponseRecord:
```

Comment:
This is a cool method, but it needs the ability to include response metadata (eg: partition keys) as well.